app/main.py

```diff
@@ -1,34 +1,103 @@
-    def __init__(self, start, end, is_drowned=False):
+    decks = []
+
+    def __init__(self,
+                 start: tuple,
+                 end: tuple,
+                 is_drowned: bool = False) -> None:
         # Create decks and save them to a list `self.decks`
-        pass
+        self.deck = self.create_deck(start, end)
+        Ship.decks.append(self)
+
+    @staticmethod
+    def create_deck(start: tuple, end: tuple) -> list:
+        if start[0] == end[0]:
+            return [(start[0], point) for point in range(start[1], end[1] + 1)]
+
+        return [(start[1], point) for point in range(start[0], end[0] + 1)]
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+class NotEnoughtShips(Exception):
+    pass
+
+
+class MissingSomeShips(Exception):
+    pass
 
 
 class Battleship:
-    def __init__(self, ships):
+    battle_field = [["~", "~", "~", "~", "~", "~", "~", "~", "~", "~"],
+                    ["~", "~", "~", "~", "~", "~", "~", "~", "~", "~"],
+                    ["~", "~", "~", "~", "~", "~", "~", "~", "~", "~"],
+                    ["~", "~", "~", "~", "~", "~", "~", "~", "~", "~"],
+                    ["~", "~", "~", "~", "~", "~", "~", "~", "~", "~"],
+                    ["~", "~", "~", "~", "~", "~", "~", "~", "~", "~"],
+                    ["~", "~", "~", "~", "~", "~", "~", "~", "~", "~"],
+                    ["~", "~", "~", "~", "~", "~", "~", "~", "~", "~"],
+                    ["~", "~", "~", "~", "~", "~", "~", "~", "~", "~"],
+                    ["~", "~", "~", "~", "~", "~", "~", "~", "~", "~"]]
+
+    def __init__(self, ships: list) -> None:
         # Create a dict `self.field`.
         # Its keys are tuples - the coordinates of the non-empty cells,
         # A value for each cell is a reference to the ship
         # which is located in it
-        pass
+        self.field = self.create_field(ships)
+        self._validate_field()
 
-    def fire(self, location: tuple):
+    def fire(self, location: tuple) -> str:
         # This function should check whether the location
         # is a key in the `self.field`
         # If it is, then it should check if this cell is the last alive
         # in the ship or not.
-        pass
+        for ship in Ship.decks:
+            self.battle_field[location[0]][location[1]] = "x"
+            if location in ship.deck and len(ship.deck) == 1:
+                Ship.decks.remove(ship)
+                return "Sunk!"
+            elif location in ship.deck:
+                ship.deck.remove(location)
+                return "Hit!"
+        return "Miss!"
+
+    @staticmethod
+    def create_field(ships: list) -> dict:
+        res = {}
+        for ship in ships:
+            res[ship] = Ship(start=ship[0], end=ship[1])
+        return res
```

Comment:
Pretty sure the task was to make dict, where key was tuple of coordinates, which makes whole code a lot more efficient, since you won't iterate through ship in finding coords, but hit coords directly... Try to refactor that, 'cause that's not according to readme, even though I understand overall logic